pkg/cloudprovider/drift.go

```diff
@@ -31,24 +31,33 @@ import (
 )
 
 func (c *CloudProvider) isNodeTemplateDrifted(ctx context.Context, machine *v1alpha5.Machine, provisioner *v1alpha5.Provisioner, nodeTemplate *v1alpha1.AWSNodeTemplate) (bool, error) {
-	ec2Instance, err := c.getInstance(ctx, machine.Status.ProviderID)
+	// nodeTemplate selectors can be nil if the user is using a launchTemplateName to define their subnets, security groups, and AMIs.
+	// Karpenter will not drift on changes if launchTemplateName is used.
+	if nodeTemplate.Spec.LaunchTemplateName != nil {
+		return false, nil
+	}
```

Comment:
Why are we moving this here? A user is required to have a subnetSelector, which will mean that the subnet status will always be filled. Also, do we know if a `launchTemplateName` is used that, karpenter will not use the default AMIs or resolved subnets